### DISCUSSION TITLE
How to route static files based on host?

### DISCUSSION THREAD

**Discussion by Yuki-Nagato** [dotnet/aspnetcore#25809] [General] [answered]
`IApplicationBuilder.UseStaticFiles` only has a `requestPath` parameter, but I want to serve static files in different folder based on different host (domain).

For example:

```
https://static-a.example.com/ -> wwwroot-a/
https://static-b.example.com/ -> wwwroot-b/
```

**Comment by Tratcher** [comment, marked as answer]
You can do that with [UseWhen](https://github.com/dotnet/aspnetcore/blob/360914cb0b2f467504a59b39464da642982c738f/src/Http/Http.Abstractions/src/Extensions/UseWhenExtensions.cs#L23).
```
app.UseWhen(context => context.Request.Host.Host.Equals("static-a.example.com", StringCompairison.OrdinalIgnoreCase), subApp => subApp.UseStaticFiles....);
```
